File: backend/src/main/java/com/gxs/bank/model/AuditLog.java
package com.gxs.bank.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "audit_logs")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AuditLog {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "performed_by", nullable = false)
    private UUID performedBy; // Employee ID

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Action action;

    @Column(name = "target_entity_type", nullable = false)
    private String targetEntityType;

    @Column(name = "target_entity_id")
    private UUID targetEntityId;

    @Column(length = 1000)
    private String details;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }

    public enum Action { KYC_APPROVED, KYC_REJECTED, CARD_APPROVED, CARD_REJECTED, LOAN_APPROVED, LOAN_REJECTED, ACCOUNT_FROZEN, ACCOUNT_UNFROZEN }
}

File: backend-python/com/gxs/bank/model/AuditLog.py
from datetime import datetime
from enum import Enum
import uuid

from sqlalchemy import Column, DateTime, Enum as SQLEnum, String, Text

from com.gxs.bank.config.database import Base


class Action(str, Enum):
    KYC_APPROVED = "KYC_APPROVED"
    KYC_REJECTED = "KYC_REJECTED"
    CARD_APPROVED = "CARD_APPROVED"
    CARD_REJECTED = "CARD_REJECTED"
    LOAN_APPROVED = "LOAN_APPROVED"
    LOAN_REJECTED = "LOAN_REJECTED"
    ACCOUNT_FROZEN = "ACCOUNT_FROZEN"
    ACCOUNT_UNFROZEN = "ACCOUNT_UNFROZEN"


class AuditLog(Base):
    __tablename__ = "audit_logs"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    performedBy = Column("performed_by", String(36), nullable=False)
    action = Column(SQLEnum(Action), nullable=False)
    targetEntityType = Column("target_entity_type", String(255), nullable=False)
    targetEntityId = Column("target_entity_id", String(36), nullable=True)
    details = Column(Text, nullable=True)
    createdAt = Column("created_at", DateTime, nullable=False, default=datetime.utcnow)


AuditLog.Action = Action
